sdks/java/io/csv/src/main/java/org/apache/beam/sdk/io/csv/CsvIOStringToCsvRecord.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.csv;
+
+import java.io.IOException;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.errorhandling.BadRecord;
+import org.apache.beam.sdk.transforms.errorhandling.ErrorHandler;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+/**
+ * {@link CsvIOStringToCsvRecord} is a class that takes a {@link PCollection<String>} input and
+ * outputs a {@link PCollection<CSVRecord>} with potential {@link PCollection<CsvIOParseError>} for
+ * targeted error detection.
+ */
+class CsvIOStringToCsvRecord
+    extends PTransform<PCollection<String>, PCollection<Iterable<String>>> {
```

Comment:
Should this be outputting `CSVRecord`?
```suggestion
    extends PTransform<PCollection<String>, PCollection<CSVRecord>> {
```

Reply:
Output was originally CSVRecord, but due to CSVRecord being read-only and formatting differences in equals() method, it is not possible to compare CSVRecord(read-only) in test checks.

Reply:
Hmmm I see. We should probably find a way around that. Looking at the `CSVRecord` class, there's some useful methods we'd be missing out on if we cast it to an `Iterable<String>`.

Instead, we can maybe try converting the output CSVRecord to an ArrayList before doing our test checks. For example:

```java
PCollection<String> outputStrings = input
        .apply(underTest)
        .apply(MapElements.into(TypeDescriptors.lists(Typedescriptors.strings())).via(
                                record -> ImmutableList.copyOf(record.iterator())));

// Passert checks with outputStrings
```

Reply:
You can implement custom comparison in test to override the equals

I.e

```
assertThat(csvRecords)
          .comparingElementsUsing(
           Correspondence.from(...customComparator) 
```

Reply:
Thank you, I will implement both measures and see the best way to go about getting around the equals mismatch. @ahmedabu98 I agree, CSVRecord is also easier to convert into custom type so overall a better choice for corresponding classes that require CsvIOStringToCsvRecord.

Reply:
Upon further documentation, in future versions the Serializable is  deprecated in CSVRecord (versions later than 1.8, which is currently being used). This was the main reason for using Iterable<String> rather than CSVRecord.

Reply:
Ahh I see, that is pretty unfortunate. I guess `Iterable<String>` makes sense to go with (even though the naming of this transform will be a little misleading)

Reply:
Naming itself isn't the best but only data truly needed from the CsvRecord is the cell for future parsing.